.codegen/workspaces.go.tmpl

```diff
@@ -30,6 +32,10 @@ func NewWorkspaceClient(c ...*Config) (*WorkspaceClient, error) {
 		// default config
 		cfg = &config.Config{}
 	}
+	if cfg.AccountID == "" || cfg.IsAccountClient() {
```

Comment:
it is not enough, proper implementation should be lazy evaluated - e.g. after the error. otherwise this will break certain terraform workflows, where account id is not yet available and is loaded from another resource